### PR TITLE
Snapshot fetch fixes

### LINKED ISSUE #2163: Support the new Protocol Labs snapshot service
**Issue summary**
We shoud update our forest-cli's `snapshot fetch` subcommand to support the new PL snapshot service: snapshots.mainnet.filops.net.

The old service (fil-chain-snapshots-fallback.s3.amazonaws.com) that we are using today is going to be deprecated at EOY.

**Task summary**
- [ ] Add support for this new service (download + validation)
- [ ] Update other subcommands if needed (ie, `prune`)

**Acceptance Criteria**
- [ ] We can fetch a new snapshot from this new service and sync from it
- [ ] Other related snapshot subcommands still work (`clean`, `list`, `prune`, `remove`)

**Other information and links**
https://pl-strflt.notion.site/pl-strflt/Lightweight-Filecoin-Chain-Snapshots-17e4c386f35c44548f5863afb7b5e024

The new naming scheme is `<height>_YYYY_MM_DDTHH_MM_SSZ.car`.

### PR DESCRIPTION
**Pull request by sudo-shashank**
**Summary of changes**
Changes introduced in this pull request:
- Fix snapshot fetch default server selection
- Optimise snapshot new pattern matching

**Reference issue to close (if applicable)**

Closes #2163 

**Other information and links**

[Support the new protocol labs snapshot service](https://github.com/ChainSafe/forest/pull/2194#top)
#2194

### CODE REVIEW

File: forest/daemon/src/daemon.rs [hunk 1]
```diff
@@ -399,7 +399,7 @@ async fn maybe_fetch_snapshot(should_fetch_snapshot: bool, config: Config) -> Co
         let path = match snapshot_fetch(
             &snapshot_path,
             &config,
-            &SnapshotServer::Filecoin,
+            &Some(SnapshotServer::Forest),
```
Comment (lemmih) [inline forest/daemon/src/daemon.rs:402]: This won't work for downloading a mainnet snapshot. How about using `None`?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: forest/daemon/src/daemon.rs (hunk 1 above)
Comment (sudo-shashank) [reply forest/daemon/src/daemon.rs]: fixed

File: forest/shared/src/cli/snapshot_fetch.rs [hunk 2]
```diff
@@ -67,31 +66,40 @@ pub struct SnapshotStore {
...
-                            let height = captures.name("height").unwrap().as_str().parse::<i64>().unwrap();
-                            let snapshot = SnapshotInfo {
-                                network,
-                                date,
-                                height,
-                                path,
-                            };
-                            snapshots.push(snapshot);
+                .map(|entry| entry.path())
+                .filter(|p| is_car_or_tmp(p))
```
Comment (creativcoder) [inline forest/shared/src/cli/snapshot_fetch.rs:75]: maybe `filter_map` instead?

**Review by lemmih — APPROVED** [review]

File: forest/shared/src/cli/snapshot_fetch.rs (hunk 2 above)
Comment (sudo-shashank) [reply forest/shared/src/cli/snapshot_fetch.rs:75]: @creativcoder not sure if `filter_map` will be right fit here
we will have to do something like this to use it
```
.filter_map(|entry| {
     if is_car_or_tmp(&entry.path()) {
        Some(entry.path())
     } else {
        None
     }
})
```

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**